addons/website_sale/controllers/main.py

```diff
@@ -228,7 +228,7 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
         attributes_ids = attributes_obj.search(cr, uid, [('attribute_line_ids.product_tmpl_id', 'in', product_ids)], context=context)
         attributes = attributes_obj.browse(cr, uid, attributes_ids, context=context)
 
-        from_currency = pool.get('product.price.type')._get_field_currency(cr, uid, 'list_price', context)
+        from_currency = pricelist.items_id[0].currency_id
```

Comment:
No,
from_currency should be taken from the current company
to_currency should be the currency of the pricelist